### ISSUE TITLE
Neovim Ruby Host race condition

### ISSUE THREAD

**Issue by stephenprater** [alexgenco/neovim-ruby#5]
I've been working on a `legacy-ruby-provider` using this plugin, and I think I've found a threading problem.

Basically, this:
```viml
let l:host = rpcstart(s:remote_host_path, [s:plugin_path])
if rpcrequest(l:host, 'poll') == 'ok
  return l:host
endif
```

Does not work - the process just hangs forever.

While this:
```viml
let l:host = rpcstart(s:remote_host_path, [s:plugin_path])
sleep 500m
if rpcrequest(l:host, 'poll') == 'ok'
  return l:host
endif
```

Works fine.

As a further data point - if I send an RPC request to the remote host that it does not respond to - (in this case anything other than 'poll') then the process hangs forever.  I looked over the code, and it appear like the EM loop isn't started until the plugins are fully loaded - but this behavior would seem to indicate otherwise.

**Comment by alexgenco** [comment]
Thanks for the info, I'm glad you're working on this. The race condition is definitely on my [radar](https://github.com/alexgenco/neovim-ruby/blob/v0.0.2/spec/acceptance/neovim-ruby-host_spec.rb#L37), but I haven't had the time to track it down yet. I wasn't aware of hanging on unknown requests, but it's possibly the same issue.

I should note that as of 0.0.2 I removed the EventMachine dependency, although the race condition was still there before I did.

**Comment by alexgenco** [comment]
I just pushed a fix for my original issue (https://github.com/alexgenco/neovim-ruby/commit/295a3b9eb41b38f7ba604849f94e196611773dee). Can you try it out on master and see if it fixed your issue as well?
I also added the "poll" request and error handling around unknown requests (https://github.com/alexgenco/neovim-ruby/commit/a50a09b2395cacba27324ba3223a9bf482f5a673)

**Comment by stephenprater** [comment]
Beautiful!  Works perfectly :+1:

**Comment by alexgenco** [comment]
Glad to hear it! Let me know if you have any other issues, otherwise I'll probably release 0.0.3 later today.

**Comment by alexgenco** [comment]
Just released 0.0.3
